lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -76,8 +78,30 @@ SmallVector<Direction> direction::unpackAttribute(Operation *component) {
 // Utilities
 //===----------------------------------------------------------------------===//
 
-// Convenience function for getting the SSA name of @p v under the scope of
-// operation @p scopeOp
+/// Verify that the value is not a "complex" value. For example, the source
+/// of an AssignOp should be a constant or port, e.g.
+/// %and = comb.and %a, %b : i1
+/// calyx.assign %port = %and, %c1_i1 ? : i1   // Incorrect
+/// calyx.assign %port = %c1_i1, %and ? : i1   // Correct
+/// TODO(Calyx): This is useful to verify current MLIR can be lowered to the
+/// native compiler. Remove this when Calyx supports wire declarations.
+/// See: https://github.com/llvm/circt/pull/1774 for context.
+template <typename Op>
+static LogicalResult verifyNotComplexSource(Op op) {
+  Operation *definingOp = op.src().getDefiningOp();
+  if (definingOp == nullptr)
+    // This is a port of the parent component.
+    return success();
+
+  if (isa<comb::XorOp, comb::AndOp, comb::OrOp>(definingOp))
```

Comment:
Can this just check if the definingOp is in the `comb` dialect?

Reply:
Sure if you show me how :-). I see Operation* has a getDialect(), but I don't think that's what I want?

Reply:
Nevermind, just using `isa` works!